### PR TITLE
osdc: invoke notify finish context on linger commit failure

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/23966
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/librados/IoCtxImpl.cc [hunk 1]
```diff
@@ -148,13 +148,11 @@ struct C_aio_notify_Complete : public C_aio_linger_Complete {
     if (ret_val == 0 && r < 0) {
       ret_val = r;
     }
+    lock.Unlock();
 
     if (acked && finished) {
-      lock.Unlock();
```
Comment (tchaikov) [inline src/librados/IoCtxImpl.cc]: `acked` and `finished` are updated with `lock`, so probably we should check them with the same lock?

Comment (dillaman) [reply src/librados/IoCtxImpl.cc]: Indeed -- you're correct. Thx.

**Comment by tchaikov** [comment]
@dillaman thanks. i think it's fine.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-05-04_16:50:59-rados-wip-kefu-testing-2018-05-04-2301-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-05-05_02:02:06-rados-wip-kefu-testing-2018-05-04-2301-distro-basic-mira/

**Pull request merged**